### PR TITLE
Make revapi report functional

### PR DESCRIPTION
**Pull request by zedbeit**
The idea is to improve revapi reporting.

Some screenshots of how the report looks like now.

img1

img3
img4
img5

### CODE REVIEW

**Review by metlos — COMMENTED** [review]
This looks ok... Could also please provide some rendered examples in the PR description so that we have a better idea about the reports?

File: independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <analysisConfiguration>
   <revapi.java>
     <missing-classes>
-      <behavior>error</behavior>
```
Comment (maxandersen) [inline independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml:6]: specific reason why changing this?

**Comment by maxandersen** [comment]
i'm fine merging this - its miles better than current one :)

just that one question on chaning from error to report I wanted to understand.

**Comment by zedbeit** [comment]
The reason is for revapi to report missing classes not to fail the build

**Comment by metlos** [comment]
Note that `report` is the default behavior. I'm not sure why `error` was put in place in Quarkus config, but the effect of `error` is essentially an exception during the analysis, which IMHO is not that useful. `report` will report the missing classes as part of the analysis results so that they can be worked with in the reports.

**Comment by metlos** [comment]
One thing that I would probably try to add is the "example use-chain" that Revapi produces for the missing classes (and other differences, see `reportUsesFor` in https://revapi.org/revapi-java/0.27.0/configuration.html). This may be helpful for figuring out why a class is considered a part of the API by Revapi.

**Comment by zedbeit** [comment]
Thank you @metlos for your inputs.

**Comment by maxandersen** [comment]
reason for fail/error was whne running pr builds the build would not pass if missing documentation for api changes.

I thought this was the way to do so ?

**Comment by maxandersen** [comment]
@zedbeit can you rebase this then i'll merge it and we can look at example-use-chain in separate pr?

**Comment by zedbeit** [comment]
Ok done.

**Comment by zedbeit** [comment]
Added changed/updated dependencies to the diff list.

This is how it now looks:

**Comment by maxandersen** [comment]
sorry did not notice the extra commits before now. Makes a big diff on the change log. awesome.

**Comment by maxandersen** [comment]
@zedbeit can you squash the commits then we can merge it?

**Comment by zedbeit** [comment]
Made a new PR for this [here,](https://github.com/quarkusio/quarkus/pull/28722) closing this one @maxandersen

**Pull request closed without merging**